src/python/pants/init/logging.py

```diff
@@ -163,6 +165,10 @@ def initialize_stdio(global_bootstrap_options: OptionValueContainer) -> Iterator
             tuple(message_regex_filters),
             log_path,
         )
+        sys.stdin = TextIOWrapper(
+            BufferedReader(raw_stdin), encoding=locale.getpreferredencoding(False)
```

Comment:
This encoding you're setting up is the default. Consider dropping unless you want to highlight a subtle point in which case consider commenting the subtle point.

Reply:
Will add a comment: if it's not set, the constructor of `TextIOWrapper` pokes the underlying file and fails, because at this point there is no actual associated destination.